server/service/src/sync/sync_on_central/mod.rs

```diff
@@ -37,13 +41,23 @@ pub async fn pull(
         batch_size,
         sync_v5_settings,
         is_initialised,
+        sync_v6_version,
     }: SyncPullRequestV6,
 ) -> Result<SyncBatchV6, SyncParsedErrorV6> {
     use SyncParsedErrorV6 as Error;
 
     if !CentralServerConfig::is_central_server() {
         return Err(Error::NotACentralServer);
     }
+
+    if !is_sync_version_compatible(sync_v6_version) {
+        return Err(Error::SyncVersionMismatch(
+            MIN_VERSION,
+            MAX_VERSION,
+            sync_v6_version,
```

Comment:
curious for thoughts - the repetition of this in each method makes me want to return this error from the `is_sync_version_compatible` and just use `?`, but I like explicitly reading through the method and seeing the different errors returned?

Making me wonder whether we have a `validate` method or something for shared checks like is_central_server, is_sync_version_compatible, credential check?

Reply:
Yeah ! We've actually [added middleware](https://github.com/msupply-foundation/open-msupply/blob/526046cd357ac9d6c8af5aa4d02692c8e8f2c8cb/server/server/src/middleware/central_server_only.rs), which i think is to complex (or maybe too verbose is the word), would rather a common validate method